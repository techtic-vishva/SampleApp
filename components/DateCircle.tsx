import React from 'react'
import { Dimensions } from 'react-native'
import Svg, { Circle } from 'react-native-svg'
import { dichotomousHippopotamus } from '../constants/Colors'

export default function DateCircle({
	progress,
	lineWidth,
	widthPercent,
	widthPixels,
	includeShadow,
	colorOverride,
	goalProgress
}: {
	progress: number
	lineWidth: number
	includeShadow?: boolean
	widthPercent?: number
	widthPixels?: number
	colorOverride?: string
	goalProgress?: number
}) {
	if (widthPercent === undefined && widthPixels === undefined) {
		throw 'Must specify either `widthPercent` or `widthPixels`.'
	}

	progress = progress * 100
	const adjustedValue = Math.min(100, goalProgress ? goalProgress : progress)
	const size = widthPixels || Dimensions.get('window').width * (widthPercent || 0)
	const r = size * 0.4
	const circumference = r * 2 * Math.PI
	return (
		<Svg
			width={size}
			height={size}
			style={[
				{ transform: [{ rotate: '270deg' }] },
				adjustedValue >= 100
					? {
							shadowColor: dichotomousHippopotamus,
							shadowOffset: { width: 0, height: 0 },
							shadowRadius: r * 0.3,
							shadowOpacity: 0.8
					  }
					: {}
			]}>
			{includeShadow !== false && (
				<Circle
					stroke={'#3d2204'}
					strokeWidth={lineWidth}
					r={r}
					cx={size / 2}
					cy={size / 2}
					strokeDasharray={[circumference]}
					strokeDashoffset={0}
				/>
			)}
			<Circle
				stroke={colorOverride || dichotomousHippopotamus}
				strokeWidth={lineWidth}
				r={r}
				cx={size / 2}
				cy={size / 2}
				strokeDasharray={[circumference]}
				strokeDashoffset={circumference - (adjustedValue / 100) * circumference}
				strokeLinecap={'round'}
			/>
		</Svg>
	)
}
