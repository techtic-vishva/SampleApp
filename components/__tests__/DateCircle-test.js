import * as React from 'react'
import renderer from 'react-test-renderer'
import DateCircle from '../DateCircle'

it('render when passing the props', () => {
	const tree = renderer
		.create(
			<DateCircle
				progress={0}
				lineWidth={3}
				widthPercent={0.1}
				goalProgress={0}
				colorOverride={'#000000'}
				includeShadow={false}
				widthPixels={50}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
