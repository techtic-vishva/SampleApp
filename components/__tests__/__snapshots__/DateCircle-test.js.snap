// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the props 1`] = `
<RNSVGSvgView
  bbHeight={50}
  bbWidth={50}
  focusable={false}
  height={50}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Array [
        Object {
          "transform": Array [
            Object {
              "rotate": "270deg",
            },
          ],
        },
        Object {},
      ],
      Object {
        "flex": 0,
        "height": 50,
        "width": 50,
      },
    ]
  }
  width={50}
>
  <RNSVGGroup
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
  >
    <RNSVGCircle
      cx={25}
      cy={25}
      propList={
        Array [
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={20}
      stroke={4278190080}
      strokeDasharray={
        Array [
          125.66370614359172,
          125.66370614359172,
        ]
      }
      strokeDashoffset={125.66370614359172}
      strokeLinecap={1}
      strokeWidth={3}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
